expression/builtin_time.go

```diff
@@ -1770,6 +1770,52 @@ func getTimeZone(ctx context.Context) *time.Location {
 	return ret
 }
 
+// isDuration returns a boolean indicating whether the str is the format of duration.
+func isDuration(str string) bool {
+	const durationArgReg = `^(|[-]?)(|\d{1,2}\s)(\d{2,3}:\d{2}:\d{2}|\d{1,2}:\d{2}|\d{1,6})(|\.\d*)$`
```

Comment:
You can pre-compile the regexp and store it as a global variable, rather than compile it every time.
Or maybe it's a non-frequency function and you can just leave as it is.